samcli/hook_packages/terraform/hooks/prepare.py

```diff
@@ -301,6 +301,124 @@ def _validate_referenced_resource_matches_sam_metadata_type(
         )
 
 
+def _get_lambda_function_source_code_path(
+    sam_metadata_attributes: dict,
+    sam_metadata_resource_address: str,
+    project_root_dir: str,
+    src_code_property_name: str,
+    property_path_property_name: str,
+    src_code_attribute_name: str,
+) -> str:
+    """
+    Validate that sam metadata resource contains the valid metadata properties to get a lambda function source code.
+    Parameters
+    ----------
+    sam_metadata_attributes: dict
+        The sam metadata properties
+    sam_metadata_resource_address: str
+        The sam metadata resource address
+    project_root_dir: str
+        the terraform project root directory path
+    src_code_property_name: str
+        the sam metadata property name that contains the lambda function source code or docker context path
+    property_path_property_name: str
+        the sam metadata property name that contains the property to get the source code value if it was provided
+        as json string
```

Comment:
why are these two customizable?

Reply:
sorry .. did not get your comment.

Reply:
discussed outside this, but I believe I was confused into why it wouldn't always be the same property name for these. It is because this function can be used for image and zip types, which will have different property names. We could potentially add some type hints for the valid values but yeah not sure how much benefit/value that brings